packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -19,41 +19,64 @@ import { first, sortBy } from "lodash";
 import { createHash } from "node:crypto";
 import { readFileSync } from "node:fs";
 
-export interface GeneratePersistedQueryManifestOptions {
-  files?: string[];
+type RequireKeys<T, K extends keyof T> = Required<Pick<T, K>> & Omit<T, K>;
 
-  globInclude?: string[];
-  globExclude?: string[];
+export interface PersistedQueryManifestConfig {
+  /**
+   * Paths to your GraphQL documents: queries, mutations, subscriptions, and fragments.
+   */
+  documents?: string | string[];
+
+  /**
+   * Paths that should be ignored when searching for your GraphQL documents.
+   */
+  documentIgnorePatterns?: string | string[];
+
+  /**
+   * Path where the manifest file will be written.
+   */
+  output?: string;
 }
 
+type DefaultPersistedQueryManifestConfig = RequireKeys<
+  PersistedQueryManifestConfig,
+  "documents" | "documentIgnorePatterns" | "output"
+>;
+
 export interface PersistedQueryManifestOperation {
   id: string;
   name: string;
   type: "query" | "mutation" | "subscription";
   body: string;
 }
+
 export interface PersistedQueryManifest {
   format: "apollo-persisted-query-manifest";
   version: 1;
   operations: PersistedQueryManifestOperation[];
 }
 
-export async function generatePersistedQueryManifest(
-  options: GeneratePersistedQueryManifestOptions = {},
-): Promise<PersistedQueryManifest> {
-  const files = new Set<string>(options.files ?? []);
-  const globInclude = options.globInclude ?? [
-    "src/**/*.{graphql,tsx,jsx,ts,js}",
-  ];
-  const globExclude = options.globExclude ?? [
+export const defaults: DefaultPersistedQueryManifestConfig = {
+  documents: "src/**/*.{graphql,js,jsx,ts,tsx}",
+  documentIgnorePatterns: [
+    "**/*.d.ts",
     "**/*.spec.{js,jsx,ts,tsx}",
     "**/*.story.{js,jsx,ts,tsx}",
     "**/*.test.{js,jsx,ts,tsx}",
-  ];
-  for (const include of globInclude) {
-    for (const f of await glob(include, { ignore: globExclude ?? [] })) {
-      files.add(f);
-    }
+  ],
+  output: "persisted-query-manifest.json",
```

Comment:
I would lean towards stdout as the default (I'm not sure there are great defaults, most likely people want things under `src`, and it's a lot easier for folks to just change stdout to a file without even using a config file than to figure out what the program's shorthand for writing to stdout is?)

Reply:
The reason I went this way is because our initial list of instructions included this command:

```
rover pq publish <GRAPH_REF> --manifest ./operations/pqs/pqm.json
```

This assumes a written file to read from. I'm fine changing it to stdout as the default if the plan is to document this command using stdout by default.

Reply:
I'm going to merge as-is for now and we can revisit this topic 🙂

Reply:
I still feel like printing to stdout is a better default.

If I have a CLI program that prints to stdout, I 100% of the time immediately know that if I'd rather it go to a file I can just use `>`.
If I have a CLI program that prints to a file, I have to figure out which convention it uses for stdout (`-`? `STDOUT`? something special?) in order to change it.

Plus it'll help remind us to take care to print any errors to stderr :)

Reply:
I respectfully disagree and I came to this decision for a couple reasons.

1. Our work-in-progress documentation discusses in detail creating persisted query manifest files, which also shows examples of the `rover` command that uses a file to generate that list in GraphOS. I felt there would be a huge disconnect if we defaulted to stdout when our documentation encourages files.
2. Files are trackable in `git`, and I expect many teams will want to see diffs in code reviews to understand what changes are made to query bodies. Writing to stdout is much more invisible and hard to understand what might be changing.

That being said, I see no reason why we couldn't add a CLI option that allows you to print the result to stdout/stderr in case people like that workflow better. I just wouldn't make this the default behavior.

Thoughts?

Reply:
Happy to agree to disagree (and leave it your way) here. But it does seem like something that would be helpful to be able to override on a per-execution basis rather than just in the config file, and with stdout as an easily achievable goal (which is I think what you're saying here).

(For example, if you have some sort of super custom transform that requires you to post-process the manifest file every time, it does seem like writing to stdout and using a pipe would be attractive.)

Reply:
Oh yes, sorry I should have mentioned that when I said "CLI option", I meant you'd do something like:

```
npx generate-persisted-query-manifest --stdout
```

(option name TBD)

This would override the `output` in your config file. I don't see this being an option you change in the config file itself.

Do you have recommendations on the option name? Not sure if there is a common convention in commands for this kind of thing or not.